apps/demos/Demos/Chat/AIAndChatbotIntegration/jQuery/index.js

```diff
@@ -0,0 +1,199 @@
+    },
+  });
+
+  const chatService = new AzureOpenAI({
+    dangerouslyAllowBrowser: true,
+    deployment,
+    endpoint,
+    apiVersion,
+    apiKey,
+  });
+
+  async function getAIResponse(messages) {
+    const params = {
+      messages: messages,
+      max_tokens: 1000,
+      temperature: 0.7,
+    };
+
+    const responseAzure = await chatService.chat.completions.create(params);
+    const data = { choices: responseAzure.choices };
+
+    return data.choices[0].message?.content;
+  };
+
+  function alertLimitReached() {
+    instance.option({
+      alerts: [{
+        message: 'Request limit reached, try again in a minute.',
+      }],
+    });
+
+    setTimeout(() => {
+      instance.option({ alerts: [] });
+    }, 10000);
+  };
+
+  async function processMessageSending() {
+    instance.option({ typingUsers: [assistant] });
+
+    try {
+      const aiResponse = await getAIResponse(messages);
+
+      setTimeout(() => {
+        instance.option({ typingUsers: [] });
+
+        messages.push({ role: 'assistant', content: aiResponse });
+
+        renderMessage(aiResponse);
+      }, 200);
+    } catch {
+      instance.option({ typingUsers: [] });
+      alertLimitReached();
+    }
+  };
+
+  async function regenerate() {
+    try {
+      const aiResponse = await getAIResponse(messages.slice(0, -1));
+
+      updateLastMessage(false, aiResponse);
+      messages.at(-1).content = aiResponse;
+    } catch {
+      updateLastMessage(false, messages.at(-1).content);
+      alertLimitReached();
+    }
+  };
+
+  function renderMessage(text) {
+    const message = {
+      id: Date.now(),
+      timestamp: new Date(),
+      author: assistant,
+      text,
+    };
+
+    customStore.push([{ type: 'insert', data: message }]);
+  };
+
+  function updateLastMessage(regenerating, text) {
+    const { items } = instance.option();
+    const lastMessage = items.at(-1);
+    const data = {
+      text: regenerating ? REGENERATION_TEXT : text,
+    };
+
+    customStore.push([{
+      type: 'update',
+      key: lastMessage.id,
+      data,
+    }]);
+  };
+
+  function convertToHtml(value) {
+    const result = unified()
+      .use(remarkParse)
+      .use(remarkRehype)
+      .use(rehypeStringify)
+      .processSync(value)
+      .toString();
+
+    return result;
+  };
+
+  const customStore = new DevExpress.data.CustomStore({
+    key: 'id',
+    load: () => {
+      const d = $.Deferred();
+
+      setTimeout(function () {
+        d.resolve([...store]);
+      });
+
+      return d.promise();
+    },
+    insert: (message) => {
+      const d = $.Deferred();
+
+      setTimeout(function () {
+        store.push(message);
+        d.resolve();
+      });
+
+      return d.promise();
+    },
+  });
+
+  const instance = $("#dx-ai-chat").dxChat({
+    dataSource: customStore,
+    reloadOnChange: false,
+    showAvatar: false,
+    showDayHeaders: false,
+    user,
+    height: 600,
```

Comment:
Are u sure about `600`? We use `710` for other chat demos

Reply:
For all the dimensions, I used the design. I have corrected it to 710. But I am plagued by the relevance of the design)

Reply:
[ead2579888a53b48af0f5f5c894493c569ee3368]